projects/dotcms-field-elements/src/components/dot-select/dot-select.e2e.ts

```diff
@@ -1,93 +1,276 @@
 import { newE2EPage, E2EElement, E2EPage } from '@stencil/core/testing';
 import { EventSpy } from '@stencil/core/dist/declarations';
 
-describe('dot-select', () => {
+const getSelect = (page: E2EPage) => page.find('select');
+const getDotLabel = (page: E2EPage) => page.find('dot-label');
+const getHint = (page: E2EPage) => page.find('.dot-field__hint');
+const getOptions = (page: E2EPage) => page.findAll('option');
+const getError = (page: E2EPage) => page.find('.dot-field__error-message');
 
+describe('dot-select', () => {
     let page: E2EPage;
     let element: E2EElement;
     let spyStatusChangeEvent: EventSpy;
     let spyValueChangeEvent: EventSpy;
 
-    describe('with params', () => {
+    describe('render CSS classes', () => {
         beforeEach(async () => {
             page = await newE2EPage();
+        });
 
-            await page.setContent(`
-            <dot-select
-                name="testName"
-                label="testLabel"
-                hint="testHint"
-                options="|,valueA|1,valueB|2"
-                value="2"
-                required-message="testErrorMsg"
-                required="true"
-                >
-            </dot-select>`);
+        describe('with data', () => {
+            it('should be invalid, touched & dirty when no option set', async () => {
+                await page.setContent(`
+                <dot-select
+                    name="testName"
+                    label="testLabel"
+                    hint="testHint"
+                    options="|,valueA|1,valueB|2"
+                    value="2"
+                    required-message="testErrorMsg"
+                    required="true">
+                </dot-select>`);
+                element = await page.find('dot-select');
+                await page.select('select', '');
+                await page.waitForChanges();
+                expect(element).toHaveClasses(['dot-invalid', 'dot-touched', 'dot-dirty']);
```

Comment:
I feel like at some point we might want to put this array in some unique place and import it in all the tests. Don't you?